pkg/utils/flow/metrics.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package flow
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	registerOnce = make(chan struct{})
+
+	flowTaskTimingsDelay    *prometheus.HistogramVec
+	flowTaskTimingsDuration *prometheus.HistogramVec
+	flowTaskTotalDuration   *prometheus.HistogramVec
+	flowTaskResults         *prometheus.CounterVec
+)
+
+// Namespace is the metric namespace for the flow library.
+const Namespace = "flow"
```

Comment:
Nit: could be unexported, I guess.
```suggestion
const metricsNamespace = "flow"
```